src/DataTable/utils/queryParams.js

```diff
@@ -377,13 +406,29 @@ function getSubFilter(
         };
   } else if (ccSelectedFilter === "isBetween") {
     return qb
-      ? qb.between(new Date(arrayFilterValue[0]), new Date(arrayFilterValue[1]))
+      ? qb.between(
+          new Date(arrayFilterValue[0]),
+          new Date(new Date(arrayFilterValue[1]).setHours(23, 59)) // set end of day for more accurate filtering
+        )
       : fieldVal => {
           return (
             dayjs(arrayFilterValue[0]).valueOf() <= dayjs(fieldVal).valueOf() &&
             dayjs(fieldVal).valueOf() <= dayjs(arrayFilterValue[1]).valueOf()
           );
         };
+  } else if (ccSelectedFilter === "notBetween") {
+    const dates = [
+      new Date(arrayFilterValue[0]),
+      new Date(new Date(arrayFilterValue[1]).setHours(23, 59)) // set end of day for more accurate filtering
+    ];
+    return qb
+      ? qb.notBetween(...dates)
+      : fieldVal => {
+          return (
+            dayjs(arrayFilterValue[0]).valueOf() <= dayjs(fieldVal).valueOf() ||
+            dayjs(fieldVal).valueOf() <= dayjs(arrayFilterValue[1]).valueOf()
```

Comment:
this doesn't seem right. If the range is [lowerLimit, upperLimit], this is saying that

`(lowerLimit < fieldValue) OR (fieldValue < upperLimit)`

It should be

`(fieldValue < lowerLimit) OR (fieldValue > upperLimit)`

NOTE: you got it right for the outsideRange though...